src/Features/CSharp/Portable/Completion/KeywordRecommenders/DelegateKeywordRecommender.cs

```diff
@@ -30,13 +30,15 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
         {
             return
                 context.IsGlobalStatementContext ||
-                ((context.IsNonAttributeExpressionContext || context.IsTypeContext) && !context.IsConstantExpressionContext) ||
+                (ValidTypeContext(context) && !context.IsConstantExpressionContext) ||
```

Comment:
can you just pull in the IsConstantExpressionContext into that?  it's only checked there and for that purpose.  then this expression is a simple sequence of 'ors'.  not `this or this or or this and not this or this or this`.

Reply:
I actually felt this was more readable.

Reply:
Pulled it out and added the using directive change.